bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/test/java/com/google/cloud/bigtable/mirroring/hbase1_x/TestMirroringTable.java

```diff
@@ -1418,4 +1441,206 @@ public void testConcurrentOpsAreNotPerformedWhenFlowControllerRejectsRequest()
     verify(primaryTable, never()).batch(ArgumentMatchers.<Row>anyList(), any(Object[].class));
     verify(secondaryTable, never()).batch(ArgumentMatchers.<Row>anyList(), any(Object[].class));
   }
+
+  @Test
+  public void testUnmatchedScannerResultQueuesAreFlushedWhenResultScannerIsClosed()
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    ScannerResultVerifier verifier =
+        spy(mismatchDetector.createScannerResultVerifier(new Scan(), 100));
+    when(mismatchDetector.createScannerResultVerifier(any(Scan.class), anyInt()))
+        .thenReturn(verifier);
+
+    Result expected1 = createResult("test1", "value1");
+    Result expected2 = createResult("test2", "value2");
+
+    ResultScanner primaryScannerMock = mock(ResultScanner.class);
+    when(primaryScannerMock.next()).thenReturn(expected1);
+    when(primaryTable.getScanner((Scan) any())).thenReturn(primaryScannerMock);
+
+    ResultScanner secondaryScannerMock = mock(ResultScanner.class);
+    when(secondaryScannerMock.next()).thenReturn(expected2);
+    when(secondaryTable.getScanner((Scan) any())).thenReturn(secondaryScannerMock);
+
+    Scan scan = new Scan();
+    ResultScanner mirroringScanner = spy(mirroringTable.getScanner(scan));
+    assertThat(mirroringScanner.next()).isEqualTo(expected1);
+    verify(mirroringMetricsRecorder, never())
+        .recordReadMismatches(any(HBaseOperation.class), eq(1));
+    waitForMirroringScanner(mirroringScanner);
+
+    verify(verifier, times(1)).verify(any(Result[].class), any(Result[].class));
```

Comment:
Is it possible to verify that mistmatch of expected1 and expected2 was reported?